lib/spack/spack/binary_distribution.py

```diff
@@ -124,10 +125,23 @@ def write_buildinfo_file(prefix, workdir, rel=False):
         dirs[:] = [d for d in dirs if d not in blacklist]
         for filename in files:
             path_name = os.path.join(root, filename)
+            if os.path.islink(path_name):
```

Comment:
Should this only be done if `rel == False`? Otherwise all the symlinks will be relativized when the tarball is built (see `make_package_relative` in `build_tarball`).

Reply:
This is how `make_package_relative` knows which links to make relative. The same thing is done with RPATHS that are made relative.

Reply:
Ah ok it was throwing me off that they were stored in the binary cache since the relative-izing was handled before creating it. However, there is a guard `relocate_package` which only uses the `'relocate_links'` entry if `rel == True`. Even though it is entirely handled before creating the cache, it could still be worthwhile to keep `'relocate_links'` around to indicate what was changed.